lib/bindConnectors/bindMessage.js

```diff
@@ -1,15 +1,98 @@
-var when             = require('when');
-var _ 				 = require('lodash');
+var when              = require('when');
+var _ 				  = require('lodash');
+const { performance } = require('perf_hooks');
+const { client, v2 }  = require('@datadog/datadog-api-client');
+
 
 const { camelCase }  = require('../utils/mout');
 var getMissingParams = require('./getMissingParams');
 var formatInputBody  = require('./formatInputBody');
 const interceptAfterSuccess = require('./interceptAfterSuccess');
+const { initialiseLogger } = require('../logger');
+const { getDatadogSecrets } = require('./getConfigs');
 
 
-module.exports = function (message, threadneedle, connectorConfig) {
+const initDataDogApi = async () => {
+	const datadogToken = await getDatadogSecrets();
+	const configurationOpts = {
+		authMethods: {
+			apiKeyAuth: datadogToken,
+		},
+
+	};
+
+	const configuration = client.createConfiguration(configurationOpts);
+
+	client.setServerVariables(configuration, {
+		site: 'datadoghq.com'
+	});
+
+	const datadogApiInstance = new v2.MetricsApi(configuration);
+
+	return { datadogApiInstance, datadogToken } ;
+};
+
+const buildDatadogMetric = (params) => {
+	const { connectorName, connectorVersion, operationName, status, runDuration } = params;
+
+	return {
+		body: {
+			series: [
+				{
+					metric: 'connector.operation.runtime.duration',
+					type: 0,
+					points: [
+						{
+							timestamp: Math.round(new Date().getTime() / 1000),
+							value: runDuration,
+						},
+					],
+					resources: [
+						{
+							'name': connectorName,
+							'type': 'connector'
+						},
+						{
+							'name': operationName,
+							'type': 'operation'
+						},
+						{
+							'name': connectorVersion,
+							'type': 'version'
+						},
+						{
+							'name': status,
+							'type': 'status'
+						}
+					]
+				},
+			],
+		},
+	};
+};
+
+const sendMetricToDatadog = (datadogInstance, metricBody) => {
+	if (!datadogInstance) return;
+
+	datadogInstance
+	.submitMetrics(metricBody)
+	.then((data) => {
+		console.log('API called successfully. Returned data: ' + JSON.stringify(data));
+	})
+	.catch((error) => { return console.error(error); });
+};
+
+const requestEnd = (startTime, params, dataDogInstance) => {
```

Comment:
unless there's some reason `requestEnd` needs to be blocking, it would be better to make this an async function, so that when it is called in the places it currently is, the rest of the code can continue while this occurs in the background